Android/src/org/droidplanner/android/fragments/mode/ModeAutoFragment.java

```diff
@@ -1,17 +1,278 @@
+                        updateWaypointSpeed(value);
+                    }
+                    break;
+
+                case AttributeEvent.AUTOPILOT_MESSAGE:
+                    String message = intent.getStringExtra(AttributeEventExtra.EXTRA_AUTOPILOT_MESSAGE);
+                    Timber.i(message);
+            }
+        }
+    };
+    private Mission mission;
+
+    private TextView waypointSpeed;
+    private ProgressBar missionProgress;
 
-public class ModeAutoFragment extends Fragment {
 
 	@Override
 	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
 		return inflater.inflate(R.layout.fragment_mode_auto, container, false);
 	}
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        view.findViewById(R.id.mc_pause).setOnClickListener(this);
+        view.findViewById(R.id.mc_restart).setOnClickListener(this);
+        view.findViewById(R.id.mc_next).setOnClickListener(this);
+        view.findViewById(R.id.mc_prev).setOnClickListener(this);
+        view.findViewById(R.id.wp_speed_plus).setOnClickListener(this);
+        view.findViewById(R.id.wp_speed_minus).setOnClickListener(this);
+        waypointSpeed = (TextView) view.findViewById(R.id.wp_speed);
+        missionProgress = (ProgressBar) view.findViewById(R.id.mission_progress);
+    }
+
+    @Override
+	public void onClick(View v) {
+		Parameters params = drone.getAttribute(AttributeType.PARAMETERS);
+        Parameter speed = params.getParameter(WPNAV_SPEED);
+        if(mission == null){
+            mission = drone.getAttribute(AttributeType.MISSION);
+        }
+		switch(v.getId()){
+			case R.id.mc_pause: {
+                drone.pauseAtCurrentLocation();
+                break;
+            }
+			case R.id.mc_restart: {
+                if(mission.getCurrentMissionItem() == mission.getMissionItems().size()){
+                    resetAutoMode();
+                }
+                msg_mission_set_current msg = new msg_mission_set_current();
+                msg.seq = 0;
+                ExperimentalApi.sendMavlinkMessage(drone, new MavlinkMessageWrapper(msg));
+                break;
+            }
+			case R.id.mc_next: {
+                msg_mission_set_current msg = new msg_mission_set_current();
+                msg.seq = mission.getCurrentMissionItem() + 1;
+                ExperimentalApi.sendMavlinkMessage(drone, new MavlinkMessageWrapper(msg));
+                break;
+            }
+			case R.id.mc_prev: {
+                msg_mission_set_current msg = new msg_mission_set_current();
+                msg.seq = mission.getCurrentMissionItem() - 1;
+                ExperimentalApi.sendMavlinkMessage(drone, new MavlinkMessageWrapper(msg));
+                break;
+            }
+			case R.id.wp_speed_plus:
+                changeWaypointSpeedBy(100.0);
```

Comment:
@squilter I have and it either does not work in auto mode or is not implemented by ArduCopter